tests/ert_tests/lsf_queue/test_lsf_driver.py

```diff
@@ -0,0 +1,128 @@
+import os
+import shutil
+from argparse import ArgumentParser
+from textwrap import dedent
+
+import pytest
+
+from ert_shared.cli import ENSEMBLE_EXPERIMENT_MODE
+from ert_shared.cli.main import run_cli
+from ert_shared.main import ert_parser
+
+
+@pytest.fixture
+def poly_case_context(tmpdir, source_root):
+    # Clone the poly_example files needed
+    shutil.copytree(
+        os.path.join(source_root, "test-data", "local", "poly_example"),
+        os.path.join(str(tmpdir), "poly_example"),
+    )
+
+    # Add the mocked lsf commands to the runpath
+    shutil.copytree(
+        os.path.join(source_root, "tests", "ert_tests", "lsf_queue", "mocked_commands"),
+        os.path.join(str(tmpdir), "poly_example", "mocked_commands"),
+    )
+
+    with tmpdir.as_cwd():
+        yield
+
+
+def test_run_mocked_lsf_queue(poly_case_context):
+    apply_customized_config(log_level="WARNING")
+    parser = ArgumentParser(prog="test_main")
+    parsed = ert_parser(
+        parser,
+        [
+            ENSEMBLE_EXPERIMENT_MODE,
+            "poly_example/poly.ert",
+        ],
+    )
+
+    run_cli(parsed)
+
+
+def test_mock_bsub_fail_random(poly_case_context):
+    """
+    Approx 50% of the submits will fail due to the random generator in the
+    created mocked bsub script. By using the retry functionality towards
+    queue-errors in job_queue.cpp we should still manage to finalize all our runs
+    before exhausting the limits
+    """
+    bsub_random_fail_name = "bsub_random_fail"
+    introduce_bsub_failures(
+        script_name=bsub_random_fail_name, percentage_successful_submits=50
+    )
+
+    apply_customized_config(mocked_bsub=bsub_random_fail_name, log_level="WARNING")
+
+    parser = ArgumentParser(prog="test_main")
+    parsed = ert_parser(
+        parser,
+        [
+            ENSEMBLE_EXPERIMENT_MODE,
+            "poly_example/poly.ert",
+        ],
+    )
+
+    run_cli(parsed)
+
+    assert True is True
```

Comment:
If not we're really out of luck! 😁

Reply:
ah, yes, forgot that one... not sure how well we should assert at this point. Either ERT has exploded or succeeded when reaching this point. Should i traverse the runpaths and verify OK files or something? Or is it sufficient just by running without error?